code/game/objects/items/devices/last_shelter.dm

```diff
@@ -0,0 +1,69 @@
+/obj/item/device/last_shelter
+	name = "Last Shelter"
+	desc = "Powerful scanner that can teleport a cruciforms of pilgrims lost in this sector of space."
+	icon = 'icons/obj/faction_item.dmi'
+	icon_state = "last_shelter"
+	item_state = "last_shelter"
+	price_tag = 20000
+	origin_tech = list(TECH_MAGNET = 5, TECH_BLUESPACE = 9, TECH_BIO = 3)
+	var/cooldown = 15 MINUTES
+	var/last_teleport = -15 MINUTES
+	var/scan = FALSE
+
+/obj/item/device/last_shelter/New()
+
+/obj/item/device/last_shelter/attack_self(mob/user)
+	if(world.time >= (last_teleport + cooldown))
+		to_chat(user, SPAN_NOTICE("The [src] scan deep space for cruciform, it's will take a while..."))
+		last_teleport = world.time
+		scan = TRUE
+		var/obj/item/weapon/implant/core_implant/cruciform/cruciform = get_cruciform()
+		if(cruciform)
+			scan = FALSE
+			user.put_in_hands(cruciform)
+			to_chat(user, SPAN_NOTICE("The [src] founded a losted cruciform in a deep space. Now fate of this disciple is in your hands"))
+		else
+			to_chat(user, SPAN_WARNING("The [src] don't find any working cruciform in a deep space. Maybe you need to try use [src] later again."))
```

Comment:
...can't find any working cruciforms in deep space. Later, you can try [src] again. ...